File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        
        String[] list = Arrays.stream(numbers).mapToObj(String::valueOf).toArray(String[]::new);
        
        Arrays.sort(list, (s1, s2) -> (s2+s1).compareTo(s1+s2));
        for (String s : list) {
            answer += s;
        }
        
        if (answer.charAt(0) == '0') {
            return "0";
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;
bool compare(int i, int j){
    return to_string(i)+to_string(j)>to_string(j)+to_string(i);
}
string solution(vector<int> numbers) {
    string answer = "";
    sort(numbers.begin(),numbers.end(),compare);
    for(int i=0;i<numbers.size();i++){
        answer+=to_string(numbers[i]);
    }
    while(answer.length()>1&&answer[0]=='0'){
        answer.erase(answer.begin());
    }
    return answer;
}